modules/util/source-formatter/src/main/resources/dependencies/replacements.json

```diff
@@ -3241,6 +3241,20 @@
 		"skipParametersValidation": true,
 		"to": "CPTaxCategoryCreateDateComparator.getInstance(param#0#)"
 	},
+	{
+		"classNames": [
+			"JournalArticle"
+		],
+		"from": "setDDMStructureKey(String paramName)",
+		"issueKey": "LPD-30559",
+		"newImports": [
+			"com.liferay.portal.kernel.service.ServiceContextThreadLocal",
+			"com.liferay.portal.kernel.util.PortalUtil",
+			"com.liferay.dynamic.data.mapping.service.DDMStructureLocalServiceUtil",
+			"com.liferay.dynamic.data.mapping.model.DDMStructure"
+		],
+		"to": "setDDMStructureId(DDMStructureLocalServiceUtil.getStructure(ServiceContextThreadLocal.getServiceContext().getScopeGroupId(), PortalUtil.getClassNameId(DDMStructure.class), param#0#).getStructureId())"
```

Comment:
@rubenssoares-br I will validate if we can use this approach before we continue, but one thing:
You could use `GroupThreadLocal` instead of the service.